### ISSUE TITLE
ion-searchbar autofocus

### ISSUE THREAD

**Issue by pickfire** [ionic-team/ionic#17745]
# Feature Request

**Ionic version:**

[x] **4.x**

**Describe the Feature Request**

Add support for `autofocus` property to ionic core for feature parity with `ion-input`. Previous patch for `v3` in https://github.com/ionic-team/ionic/pull/8232 but probably outdated.

**Describe Preferred Solution**

Boolean property for `ion-searchbar` will do.

```html
<ion-searchbar autofocus></ion-searchbar>
```

**Describe Alternatives**

The alternative uses a simple timeout (don't follow this, not tested) based on angular.

```html
<ion-seachbar #autofocus></ion-searchbar>
```

```typescript
@Component(...)
export class SimpleSearch implements OnInit {
  @ViewContent('#autofocus') searchbar;

  ngOnInit() {
    setTimeout(() => this.searchbar.focus(), 300);
  }
}
```

**Related Code**

**Additional Context**

https://github.com/ionic-team/ionic/pull/8232

**Comment by UTSOURCE** [comment]
`@ViewChild(IonSearchbar) myInput: IonSearchbar;`
` setTimeout(() => {
        this.myInput.setFocus();
      }, 150);`

**Comment by michaeltintiuc** [comment]
Same deal with Vue, this works:

```js
mounted() {
      setTimeout(() => this.$refs.searchbar.setFocus(), 1)
}
```

**Comment by carlesferreiro** [comment]
Did anyone noticed that the alternative does not work in iOS since iOS 12.2? in 12.1 works fine but in 12.2 it does not work in safari or as app with cordova.

After calling .focus() nothing happens, but if I switch to another app and then back to the ionic app it focuses and the keyboard is shown.

**Comment by qhoirulanwar** [comment]
Ionic version:
**[x] 4.x**
Angular version:
**[x] 8.x**

`<ion-searchbar #autofocus></ion-searchbar>`

`import { Component, ViewChild } from '@angular/core';`
`import { IonSearchbar } from '@ionic/angular';`

`@ViewChild('autofocus', { read: false, static: true }) searchbar: IonSearchbar;`

`ngOnInit(){
    setTimeout(() => this.searchbar.setFocus(), 500);
  }`

**Comment by zer09** [comment]
Without setTimeout

```
<ion-searchbar #searchbar></ion-searchbar>

@ViewChild('searchbar', { static: false }) searchbar: IonSearchbar;

ionViewDidEnter() {
  this.searchbar.setFocus();
}
```

**Comment by ihsanberahim** [comment]
Like @zen09 said. I just add on.

`ViewChild` query result completed in (ngAfterViewInit) `AfterViewInit`. So, can do like this.

```
    ngAfterViewInit() {
        if (this.baseSvc.isBrowser) { // OPTIONAL: this is my condition for my ng ssr
            this.searchbar.setFocus();
        }
    }
```

**Comment by jayordway** [comment]
I am still seeing this issue with angular 9 and ionic 5, the setFocus() method is not working within any stage of the lifecycle.

**Comment by sapabg** [comment]
Yes, no matter witch lifecycle I use it will not work without some timeout.
I tried getInputElement() with .focus() - only works with timeout too.
The searchbar is present when I console.log it but the methods does not work without timeout...

**Comment by 3zzy** [comment]
I can't get this to work in Vue 3 / Ionic 6 either. 

```
<ion-searchbar ref="searchbar"></ion-searchbar>
...
const searchbar = ref(null);
onMounted(() => {
    setTimeout(() => {
        searchbar.value.setFocus(); // TypeError: ... is not a function
    }, 1000);
});
```

What am I doing wrong? How do I use the `setFocus` method in Vue? Thanks

**Comment by sapabg** [comment]
> I can't get this to work in Vue 3 / Ionic 6 either.
> 
> ```
> <ion-searchbar ref="searchbar"></ion-searchbar>
> ...
> const searchbar = ref(null);
> onMounted(() => {
>     setTimeout(() => {
>         searchbar.value.setFocus(); // TypeError: ... is not a function
>     }, 1000);
> });
> ```
> 
> What am I doing wrong? How do I use the `setFocus` method in Vue? Thanks

<ion-searchbar #searchbar></ion-searchbar>

and in the component @ViewChild('searchbar') searchbar: IonSearchbar;
than you can use searchbar.setFocus()

**Comment by iplanwebsites** [comment]
@keygun-development it's the only solution I found in vue3+ionic7. Thanks for sharing

**Comment by liamdebeasi** [comment]
Hi everyone,

The team recently did some investigation into the reliability of `autofocus` in other components such as `ion-input`. We found that mobile platforms apply various restrictions to how `autofocus` works to the point that it is pretty unreliable in mobile applications. Due to these restrictions, we think that adding an `autofocus` property to `ion-searchbar` would provide for a poor developer experience.

We've created new documentation that shows how to manage focus in a mobile application: https://ionicframework.com/docs/developing/managing-focus This can be applied to `ion-searchbar` to get the desired results. I am going to close this, but let me know if there are any questions.
